Sources/NestTestHelpers/MockFileSystem.swift

```diff
@@ -29,44 +29,52 @@ public final class MockFileSystem: FileSystem, Sendable {
         withIntermediateDirectories createIntermediates: Bool,
         attributes: [FileAttributeKey: Any]?
     ) throws {
-        if createIntermediates {
-            let components = url.pathComponents
-            var currentComponents: [String] = []
-            for component in components {
-                currentComponents.append(component)
-                if item.item(components: currentComponents) != nil {
-                    continue
+        lockedItem.withLock { item in
```

Comment:
[Author Note] 
I noticed this class `MockFileSystem` can make race condition related it `item`.
To resolve the issue, I used `withLock` function in functions using `item`.